File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/UnknownLaoException.java
package com.github.dedis.popstellar.utility.error;

import com.github.dedis.popstellar.R;

/** An exception that indicates the lao with the specified id is not known to the app */
public class UnknownLaoException extends GenericException {

  public UnknownLaoException(String laoId) {
    super("Lao with id " + laoId + " is unknown");
  }

  public UnknownLaoException() {
    super("Could not find a valid Lao");
  }

  @Override
  public int getUserMessage() {
    return R.string.unknown_lao_exception;
  }

  @Override
  public Object[] getUserMessageArguments() {
    return new Object[0];
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/UnknownLaoException.kt
package com.github.dedis.popstellar.utility.error

import com.github.dedis.popstellar.R

/** An exception that indicates the lao with the specified id is not known to the app */
class UnknownLaoException : GenericException {
  constructor(laoId: String) : super("Lao with id $laoId is unknown")

  constructor() : super("Could not find a valid Lao")

  override val userMessage: Int
    get() = R.string.unknown_lao_exception

  override val userMessageArguments: Array<Any?>
    get() = arrayOfNulls(0)
}
